calc/src/mechanics/util.ts

```diff
@@ -356,6 +357,34 @@ export function getFinalDamage(
   return OF16(pokeRound(Math.max(1, OF32(damageAmount * finalMod) / 0x1000)));
 }
 
+/**
+ * Determines which move category Shell Side Arm should behave as. Applications
+ * that consume this library may want to import this function to show users move
+ * category information.
+ *
+ * Notes:
```

Comment:
@Marty-D / @DaWoblefet

Reply:
Kris pointed me at https://github.com/smogon/pokemon-showdown/commit/65d2bb5d0c36b61a2577382cc30aa861521113fd#diff-706ad2c9c5d1920e4245d5debe9e3c11 commit. I believe the follow are the links to the research:

- https://www.smogon.com/forums/threads/sword-shield-battle-mechanics-research.3655528/page-50#post-8509843
- https://www.smogon.com/forums/threads/sword-shield-battle-mechanics-research.3655528/page-51#post-8511267
- https://www.smogon.com/forums/threads/sword-shield-battle-mechanics-research.3655528/page-51#post-8514316
- https://www.smogon.com/forums/threads/sword-shield-battle-mechanics-research.3655528/page-51#post-8514607

(man, how awesome are our researchers??)

Reply:
Thanks for digging these up for me!

I _do_ think it's a pretty safe assumption that the comparison is based on real predicted damage.  But I can't help noticing that it still doesn't look like anyone has tested Shell Side Arm with an attacker/defender with stats something like:

Attacker:
Atk 100
SpA 50

Defender:
Def 50
SpD 25

I can't think of any other way to rule out that the move choice is based on a flat _difference_ of attack stat minus defense stat.

Reply:
> Damage cannot be stochastic. You have two choices here:
> 
> - either simply favor one over the other and leave a comment explaining this decision = the approach ajhyndman originally went with
> - add a field to Move to configure how the tie should be broken (so that it could be wired up to a UI which allows the user to decide) = more correct, but potentially overkill

There is already a UI to override the category of a Move which you could use as your fallback.

> I suppose theres also hidden option c
> 
> - run the calculation twice with SSA being Physical once and Special in the other and report on both results

Ah, but the category doesn't depend on the whole calculation, only just this one part.